kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -48,31 +50,96 @@ public class StructField {
           METADATA_ROW_INDEX_COLUMN_NAME,
           LongType.LONG,
           false,
-          FieldMetadata.builder().putBoolean(IS_METADATA_COLUMN_KEY, true).build());
+          FieldMetadata.builder().putBoolean(IS_METADATA_COLUMN_KEY, true).build(),
+          Collections.emptyList());
 
   public static final String COLLATIONS_METADATA_KEY = "__COLLATIONS";
 
   ////////////////////////////////////////////////////////////////////////////////
   // Instance Fields / Methods
   ////////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * Represents a type change for a field, containing the original and new primitive types.
+   *
+   * <p>Type changes are actually persisted in metadata attached to StructFields but the rules for
+   * where the metadata is attached depend on if the change is for nested arrays/maps or primitive
+   * types.
+   */
+  public static class TypeChange {
```

Comment:
may be pull this out, given this is going to be a public API

Reply:
done